### ISSUE TITLE
My Subscriptions not loading

### ISSUE THREAD

**Issue by musicmagix** [anxdpanic/plugin.video.youtube#1032]
### Context

Please provide any relevant information about your setup

* Add-on Version: 7.1.1.6
* Kodi Version: 18.9
* Kodi GUI Language: English
* Operating System: MIUI TV 1.3.105
* Operating System Language: English

------

### Expected Behavior

Clicking on „My Subscriptions“ shows my subscriptions.

------

### Current Behavior

Clicking on „My Subscriptions“ results in a weird loading-loop during which the number of items to be loaded exceeds the total number of items and the progress bar remains at 0% while saying „Please wait…“. Occasionally the amount of total items would fluctuate between two numbers as well.

------

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Open Kodi
2. Open YT Plugin
3. Attempt to open „My Subscriptions“

------

### Log

Please include a complete [debug log](https://paste.kodi.tv/fiwoseluko).

------

### Additional Information

During the last minor updates I could, occasionally, access „My Subscriptions“ after executing the „Setup Wizard“ without changing any value. I did not even have to „Refresh settings.xml“ in step 9 (which keeps failing anyway). 

I am aware that the Kodi 18 plugin version causes trouble and extra work for the maintainer! This is meant to be helpful for other recent cases of funny behaviour of the „My Subscriptions“ section. Maybe there is an easy fix for it? It would be much appreciated as I cannot update Kodi to any newer version on my Wemax One Pro projector. (I’ve tried every major new version.)

Thank you for your ongoing work. It‘s really a unique project!

------

**Comment by musicmagix** [comment]
I just managed to access „My Subscriptions“ by first triggering the loading loop and then using the aforementioned workaround of executing the Setup Wizard. I couldn‘t get this to work at least 10 times in the last days. Updated Log: https://paste.kodi.tv/kukazucimu

**Comment by musicmagix** [comment]
https://paste.kodi.tv/xuyamatace Is the complete debug log including a successful loading of „My Subscriptions“ on the first try. (I completely closed Kodi and cleared the internal memory for this. I have no clue why it opened now.)
🫣

**Comment by MoojMidge** [comment]
> I am aware that the Kodi 18 plugin version causes trouble and extra work for the maintainer!

It's mainly an issue when logs aren't provided or when the underlying CPython implementation has changed, and in this case the latter is true.

Regardless I can see what may cause the issue, but won't be in a position to do anything about it, if it is feasible to fix, for a couple of weeks.

I can make some changes for you to try, but you will be a guinea pig as the changes will be completely untested.

**Comment by musicmagix** [comment]
`I can make some changes for you to try, but you will be a guinea pig as the changes will be completely untested.`

It will be my pleasure.

**Comment by MoojMidge** [comment]
See how you go with https://github.com/MoojMidge/plugin.video.youtube/releases/download/v7.2.0%2Bbeta.2/plugin.video.youtube-7.2.0+beta.2.leia.1.zip

**Comment by musicmagix** [comment]
Works right away!

I'll try every function in the following days but My Subscriptions is loading and videos are playing.

🙏

**Comment by MoojMidge** [comment]
> Works right away!

Promising, but not necessarily definitive because it sometimes worked without the changes as well.

I don't think the changes will cause problems, but would be good to know it has actually fixed the problem you were reporting.

Are you able to replicate the original problem using v7.1.1.6, then without trying to workaround it, update to the test version linked above and then confirm that My Subscriptions loads correctly?

**Comment by musicmagix** [comment]
Hi again,
Sorry for the wait. I‘m not sure I get you right here: I can roll the plugin back to 7.1.1.6, sure, confirm that my subscriptions is not loading and then update back to the beta and confirm it’s loading. Is that what you meant?

in the past few days I could mainly use the plugin (beta) as intended. The loading of My Subscriptions takes quite a while now. 20 or so sec. for .123 channels altogether. I did have to restart/delete from RAM Kodi occasionally. Sometimes, maybe 1 in 10, I would also be greeted with „Stream not found“. Both behaviours I cannot replicate reliably. Both disappear after restarting the plugin.

Do you want me to try the rollback?

**Comment by MoojMidge** [comment]
> I can roll the plugin back to 7.1.1.6, sure, confirm that my subscriptions is not loading and then update back to the beta and confirm it’s loading. Is that what you meant?
> Do you want me to try the rollback?

Yes, that is what I mean. You don't have to do this, but it would be helpful to confirm that the changes have actually fixed the problem.

> The loading of My Subscriptions takes quite a while now. 20 or so sec. for .123 channels altogether.

Is this taking much longer than in previous versions? How long did it take previously? When you first open My Subscriptions in v7.1.1.6 how many videos do you see in the first page? Can you compare this to how many videos you see on the first page of My Subscriptions in the beta version?

> Sometimes, maybe 1 in 10, I would also be greeted with „Stream not found“

When you see this again, can you make a note of the video, try playing it again and get a log?

**Comment by musicmagix** [comment]
> Yes, that is what I mean. You don't have to do this, but it would be helpful to confirm that the changes have actually fixed the problem.

I rolled back to 7.1.1.6, failed to load „My Subscriptions“. I then installed your beta and tried again. My Subscriptions loaded. (The failed loading cycle from 7.1.1.6 stayed visible in the menus, though. I know that it disappears once I completely close Kodi, but I wanted to keep everything in one log: [https://paste.kodi.tv/datacuzuxe](https://paste.kodi.tv/datacuzuxe)
![IMG_5167](https://github.com/user-attachments/assets/4b180256-0e93-423c-a91b-bc56c72608d5)

> > The loading of My Subscriptions takes quite a while now. 20 or so sec. for .123 channels altogether.
> 
> Is this taking much longer than in previous versions? How long did it take previously? When you first open My Subscriptions in v7.1.1.6 how many videos do you see in the first page? Can you compare this to how many videos you see on the first page of My Subscriptions in the beta version?

The time is hard to estimate, given that 7.1.1.6 (and 4-5 minor versions before) did not work for me and I stopped updating once I had found a working version in March or so. I would guess that it took around 10-15 seconds in the last working version of the update and is taking about twice as long now. 

Interestingly, the progress seems to be counting from 0-100% twice before showing the results in the beta version. (The first run stops at 95 items and 100%, the second run stops at 122 items and 100%, which is the correct number of channels. The second run takes longer.)

The number of items displayed on one page remained the same at 30.

> > Sometimes, maybe 1 in 10, I would also be greeted with „Stream not found“
> 
> When you see this again, can you make a note of the video, try playing it again and get a log?

I will!

**Comment by musicmagix** [comment]
I just recorded an instance of "Stream not found". When trying a second time, the video opened. https://paste.kodi.tv/onilitocux

**Comment by MoojMidge** [comment]
> I rolled back to 7.1.1.6, failed to load „My Subscriptions“. I then installed your beta and tried again. My Subscriptions loaded. (The failed loading cycle from 7.1.1.6 stayed visible in the menus, though. I know that it disappears once I completely close Kodi, but I wanted to keep everything in one log: https://paste.kodi.tv/datacuzuxe

Great, thanks for confirming.

> Interestingly, the progress seems to be counting from 0-100% twice before showing the results in the beta version.

That is why it is taking about twice as long - it is actually doing twice as much to properly populate the listing.

When you scroll to the end of the page does it say `Next page (2)` or `Next page (3)`?

> I just recorded an instance of "Stream not found". When trying a second time, the video opened. https://paste.kodi.tv/onilitocux

That is quite odd. Don't have the time to look into it at the moment though as it quite a difficult issue to investigate. I'll move it to a new issue so I don't forget for later.

**Comment by musicmagix** [comment]
> When you scroll to the end of the page does it say `Next page (2)` or `Next page (3)`?

@MoojMidge 
"Next Page (2)", as expected.

**Comment by MoojMidge** [comment]
> "Next Page (2)", as expected.

Then that means it should only be loading a single page of items. I'll have to have a closer look at your log to see why it appears to be loading twice.

**Comment by musicmagix** [comment]
> > "Next Page (2)", as expected.
> 
> Then that means it should only be loading a single page of items. I'll have to have a closer look at your log to see why it appears to be loading twice.

That's much appreciated. I'll be happy to assist within my abilities.

The first loading cycle has a varying number of items, usually in the 60s to 90s while the second one goes through the full 122.

**Comment by MoojMidge** [comment]
> The time is hard to estimate, given that 7.1.1.6 (and 4-5 minor versions before) did not work for me and I stopped updating once I had found a working version in March or so. I would guess that it took around 10-15 seconds in the last working version of the update and is taking about twice as long now.

Can you identify the previous version that worked for you and didn't have the slowdown you are seeing now? 

Can't see anything wrong in your logs or in the code for the new versions that could be causing the slowdown.

The issue with the double progress dialogs was just cosmetic, and should be fixed now:
https://github.com/MoojMidge/plugin.video.youtube/archive/refs/heads/master.zip

**Comment by musicmagix** [comment]
> Can you identify the previous version that worked for you and didn't have the slowdown you are seeing now?

It‘s very difficult to do that as I had been using the same version for years until it stopped working. I then found my way here and startet updating the app manually. I think the version which held up all this time must‘ve been the one on Kodi Repo, which is (currently) 6.8.25 (by anxdpanic, bromix). I can‘t tell you if or when the repo was last updated.

> Can't see anything wrong in your logs or in the code for the new versions that could be causing the slowdown.

In the current beta 4 version the loading time seems to be a bit quicker, the percentages show lots of odd jumps, though.

> The issue with the double progress dialogs was just cosmetic, and should be fixed now: https://github.com/MoojMidge/plugin.video.youtube/archive/refs/heads/master.zip

I can‘t install this version due to a missing dependency (python 3.0.0 if I read it correctly). Is this the Leia Version of the plugin?

PS: I have no reference point for „normal“ loading times. Comparisons with my usage years ago is virtually impossible as I have, for sure, added numerous channels to my subscriptions in the meantime. Given the dated hard- and software I‘m ok with the state of beta 4 as it does what it‘s supposed to. Feel free to dig further if it‘s in anyway meaningful to you. I‘ll be happy to test anything you throw at me. But for this specific issue I don‘t feel like it‘s worth the effort.

**Comment by MoojMidge** [comment]
> In the current beta 4 version the loading time seems to be a bit quicker, the percentages show lots of odd jumps, though.

Yes, there are issues with the progress dialogs due to trying to maintain compatibility with Kodi 18

> I can‘t install this version due to a missing dependency (python 3.0.0 if I read it correctly). Is this the Leia Version of the plugin?

No it wasn't, try this instead:
https://github.com/MoojMidge/plugin.video.youtube/releases/download/v7.2.0%2Bbeta.5/plugin.video.youtube-7.2.0+beta.5.leia.1.zip

> Feel free to dig further if it‘s in anyway meaningful to you. I‘ll be happy to test anything you throw at me. But for this specific issue I don‘t feel like it‘s worth the effort.

I can't find anything obvious, so, without a reference to compare, to lets call this resolved.
